src/counts.jl

```diff
@@ -238,12 +238,65 @@ If a weighting vector `wv` is specified, the sum of the weights is used rather t
 raw counts.
 """
 function addcounts!(cm::Dict{T}, x::AbstractArray{T}) where T
+    # if it's safe to be sorted using radixsort then it should be faster
+    # albeit using more RAM
+    if radixsort_safe(T)
+        addcounts_radixsort!(cm, x)
+        return cm
+    else
+        addcounts_dict!(cm,x)
+        return cm
+    end
+end
+
+"""Dict-based addcounts method"""
+function addcounts_dict!(cm::Dict{T}, x::AbstractArray{T}) where T
     for v in x
-        cm[v] = get(cm, v, 0) + 1
+        index = Base.ht_keyindex2(cm, v)
+        if index > 0
+            @inbounds cm.vals[index] += 1
+        else
+            @inbounds Base._setindex!(cm, 1, v, -index)
+        end
     end
     return cm
 end
 
+const BASE_RADIXSORT_SAFE_TYPE = Union{
+    Int8, Int16, Int32, Int64, Int128,
```

Comment:
I think the convention is rather `Union{Int8, Int16, Int32, Int64, Int128,`, and then align following lines with `Int8`. Also, type aliases are written in CamelCase.